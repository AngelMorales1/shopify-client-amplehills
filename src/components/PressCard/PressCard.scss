@import 'settings', 'typography';

.PressCard {
  border: spacing(0.7) solid color('light-gray');
  width: spacing(76);
  min-height: spacing(96);
  border-radius: radius('s');

  &__logo {
    width: spacing(25);
    height: spacing(14);
    object-fit: contain;
  }

  &__quote {
    font-size: $font-5;
    @media ($medium-up) {
      font-size: spacing(5.2);
    }
  }

  &__button {
    font-size: $font-2;
    letter-spacing: spacing(0.2);
  }
}
