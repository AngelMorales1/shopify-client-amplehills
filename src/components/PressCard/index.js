import React from 'react';
import PropTypes from 'prop-types';
import get from 'utils/get';
import cx from 'classnames';

import { Image, Button } from 'components/base';
import styles from './PressCard.scss';
import imageModel from 'models/imageModel';

const PressCard = ({ pressCard }) => {console.log(pressCard)
  return (
    <div
      className={cx(
        styles['PressCard'],
        'm2 pt4 pb3 px3 flex flex-column justify-center items-center'
      )}
    >
      <Image
        className={cx(styles['PressCard__logo'])}
        src={get(pressCard, 'logoImage.data', '')}
        alt={`${pressCard.title} logo`}
      />
      <span
        className={cx(
          styles['PressCard__quote'],
          'carter text-peach center py3 mb4'
        )}
      >{`"${pressCard.quote}"`}</span>
      <Button
        className={cx(styles['PressCard__button'], 'uppercase')}
        to={pressCard.linkUrl}
        label="Read about it"
        variant="primary-small"
        color="peach"
      />
    </div>
  );
};

PressCard.propTypes = {
  pressBlock: PropTypes.shape({
    fields: PropTypes.shape({
      logoImage: imageModel.propTypes,
      linkUrl: PropTypes.string,
      quote: PropTypes.string,
      title: PropTypes.string
    })
  })
};

PressCard.defaultProps = {
  pressBlock: {
    fields: {
      logoImage: imageModel.default,
      linkUrl: '',
      quote: '',
      title: ''
    }
  }
};

export default PressCard;
