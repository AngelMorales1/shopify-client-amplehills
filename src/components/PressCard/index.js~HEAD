import React from 'react';
import PropTypes from 'prop-types';
import get from 'utils/get';
import cx from 'classnames';

import { Image, Button } from 'components/base';
import styles from './PressCard.scss';

const PressCard = ({ pressCard }) => {

  const title = get(pressCard, 'title', '');

  return (
    <div
      className={cx(
        styles['PressCard'],
        'm2 pt4 pb3 px3 flex flex-column justify-center items-center'
      )}
    >
      <Image
        className={cx(styles['PressCard__logo'])}
        src={get(pressCard, 'logoImage.data', '')}
        alt={`${title} logo`}
      />
      <span
        className={cx(
          styles['PressCard__quote'],
          'carter text-peach center py3 mb4'
        )}
      >
        {pressCard.quote}
      </span>
      <Button
        className={cx(styles['PressCard__button'], 'uppercase')}
        to={pressCard.linkUrl}
        label="Read about it"
        variant="primary-small"
        color="peach"
      />
    </div>
  );
};

PressCard.propTypes = {
  pressCard: PropTypes.shape({
    logoImage: PropTypes.shape({
      data: PropTypes.string
    }),
    linkUrl: PropTypes.string,
    quote: PropTypes.string,
    title: PropTypes.string
  })
};

PressCard.defaultProps = {
  pressCard: {
    logoImage: {
      data: ''
    },
    linkUrl: '',
    quote: '',
    title: ''
  }
};

export default PressCard;
