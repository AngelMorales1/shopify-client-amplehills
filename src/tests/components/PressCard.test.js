import React from 'react';
import { shallow } from 'enzyme';

import PressCard from 'components/PressCard';

it('renders without data', () => {
  const component = shallow(<PressCard />);

  expect(component).toMatchSnapshot();
});

it('renders with data', () => {
  const component = shallow(
    <PressCard
      pressCard={{
        logoImage: {
          data: ''
        },
        linkUrl: '',
        quote: '',
        title: ''
      }}
    />
  );

  expect(component).toMatchSnapshot();
});
