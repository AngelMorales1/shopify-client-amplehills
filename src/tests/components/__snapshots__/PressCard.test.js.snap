// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PressCard
    pressCard={
        Object {
            "linkUrl": "",
            "logoImage": Object {
              "data": "",
            },
            "quote": "",
            "title": "",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Image
          alt=" logo"
          bg={false}
          className=""
          onImgLoad={[Function]}
          src=""
          style={Object {}}
/>,
        <span
          className="carter text-peach center py3 mb2"
>
          “”
</span>,
        <Button
          childrenWrapperClassName=""
          className="uppercase"
          color="peach"
          disabled={false}
          fullWidth={false}
          hover="shadow"
          label="Read about it"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          shadow={false}
          to=""
          type="button"
          variant="primary-small"
/>,
      ],
      "className": "m2 pt4 pb3 px3 flex flex-column justify-between items-center",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "alt": " logo",
          "bg": false,
          "children": null,
          "className": "",
          "onImgLoad": [Function],
          "src": "",
          "style": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "“”",
          "className": "carter text-peach center py3 mb2",
        },
        "ref": null,
        "rendered": "“”",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": null,
          "childrenWrapperClassName": "",
          "className": "uppercase",
          "color": "peach",
          "disabled": false,
          "fullWidth": false,
          "hover": "shadow",
          "label": "Read about it",
          "onClick": [Function],
          "onMouseEnter": [Function],
          "onMouseLeave": [Function],
          "shadow": false,
          "to": "",
          "type": "button",
          "variant": "primary-small",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Image
            alt=" logo"
            bg={false}
            className=""
            onImgLoad={[Function]}
            src=""
            style={Object {}}
/>,
          <span
            className="carter text-peach center py3 mb2"
>
            “”
</span>,
          <Button
            childrenWrapperClassName=""
            className="uppercase"
            color="peach"
            disabled={false}
            fullWidth={false}
            hover="shadow"
            label="Read about it"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            shadow={false}
            to=""
            type="button"
            variant="primary-small"
/>,
        ],
        "className": "m2 pt4 pb3 px3 flex flex-column justify-between items-center",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alt": " logo",
            "bg": false,
            "children": null,
            "className": "",
            "onImgLoad": [Function],
            "src": "",
            "style": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "“”",
            "className": "carter text-peach center py3 mb2",
          },
          "ref": null,
          "rendered": "“”",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": null,
            "childrenWrapperClassName": "",
            "className": "uppercase",
            "color": "peach",
            "disabled": false,
            "fullWidth": false,
            "hover": "shadow",
            "label": "Read about it",
            "onClick": [Function],
            "onMouseEnter": [Function],
            "onMouseLeave": [Function],
            "shadow": false,
            "to": "",
            "type": "button",
            "variant": "primary-small",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders without data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PressCard
    pressCard={
        Object {
            "linkUrl": "",
            "logoImage": Object {
              "data": "",
            },
            "quote": "",
            "title": "",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Image
          alt=" logo"
          bg={false}
          className=""
          onImgLoad={[Function]}
          src=""
          style={Object {}}
/>,
        <span
          className="carter text-peach center py3 mb2"
>
          “”
</span>,
        <Button
          childrenWrapperClassName=""
          className="uppercase"
          color="peach"
          disabled={false}
          fullWidth={false}
          hover="shadow"
          label="Read about it"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          shadow={false}
          to=""
          type="button"
          variant="primary-small"
/>,
      ],
      "className": "m2 pt4 pb3 px3 flex flex-column justify-between items-center",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "alt": " logo",
          "bg": false,
          "children": null,
          "className": "",
          "onImgLoad": [Function],
          "src": "",
          "style": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "“”",
          "className": "carter text-peach center py3 mb2",
        },
        "ref": null,
        "rendered": "“”",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": null,
          "childrenWrapperClassName": "",
          "className": "uppercase",
          "color": "peach",
          "disabled": false,
          "fullWidth": false,
          "hover": "shadow",
          "label": "Read about it",
          "onClick": [Function],
          "onMouseEnter": [Function],
          "onMouseLeave": [Function],
          "shadow": false,
          "to": "",
          "type": "button",
          "variant": "primary-small",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Image
            alt=" logo"
            bg={false}
            className=""
            onImgLoad={[Function]}
            src=""
            style={Object {}}
/>,
          <span
            className="carter text-peach center py3 mb2"
>
            “”
</span>,
          <Button
            childrenWrapperClassName=""
            className="uppercase"
            color="peach"
            disabled={false}
            fullWidth={false}
            hover="shadow"
            label="Read about it"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            shadow={false}
            to=""
            type="button"
            variant="primary-small"
/>,
        ],
        "className": "m2 pt4 pb3 px3 flex flex-column justify-between items-center",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alt": " logo",
            "bg": false,
            "children": null,
            "className": "",
            "onImgLoad": [Function],
            "src": "",
            "style": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "“”",
            "className": "carter text-peach center py3 mb2",
          },
          "ref": null,
          "rendered": "“”",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": null,
            "childrenWrapperClassName": "",
            "className": "uppercase",
            "color": "peach",
            "disabled": false,
            "fullWidth": false,
            "hover": "shadow",
            "label": "Read about it",
            "onClick": [Function],
            "onMouseEnter": [Function],
            "onMouseLeave": [Function],
            "shadow": false,
            "to": "",
            "type": "button",
            "variant": "primary-small",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
